File: configlibrary/src/main/java/com/shmedo/core/enums/LogOutputStatus.java
package com.shmedo.core.enums;

/**
 * 项目名：  mCloudapp <br/>
 * 包名：    com.shmedo.core.enums <br/>
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/4/23 <br/>
 * 描述：   日志输出状态
 */
public enum LogOutputStatus {
    CLOSE(0), //关闭

    OPEN(1);//打开


    private int status;

    LogOutputStatus(int i) {
        this.status = i;
    }

    public int toInt() {
        return status;
    }

    public static LogOutputStatus valueOf(int status) {
        switch (status) {
            case 0:
                return CLOSE;

            case 1:
                return OPEN;

            default:
                return CLOSE;
        }
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/md_cmd/enums/LogOutputStatus.kt
package com.shmedo.lib.device.base.md_cmd.enums

/**
 * 项目名：  mCloudapp <br></br>
 * 包名：    com.shmedo.core.enums <br></br>
 * 创建者:   gonghe <br></br>
 * 创建时间:  2020/4/23 <br></br>
 * 描述：   日志输出状态
 */
enum class LogOutputStatus(//打开
    private val status: Int
) {
    CLOSE(0),

    //关闭
    OPEN(1);

    fun toInt(): Int {
        return status
    }

    companion object {
        fun value(status: Int): LogOutputStatus {
            return when (status) {
                0 -> CLOSE
                1 -> OPEN
                else -> CLOSE
            }
        }
    }
}
